numba/cuda/cudadecl.py

```diff
@@ -119,17 +119,16 @@ def generic(self, args, kws):
             return signature(ary.dtype, ary, idx, ary.dtype)
 
 
-@intrinsic
-class Cuda_atomic_max(AbstractTemplate):
-    key = cuda.atomic.max
-
+class Cuda_atomic_maxmin(AbstractTemplate):
     def generic(self, args, kws):
         assert not kws
         ary, idx, val = args
-
-        # Implementation presently supports float64 only,
+        # Implementation presently supports:
+        # float64, float32, int32, int64, uint32, uint64 only,
         # so fail typing otherwise
-        if ary.dtype != types.float64:
+        supported_types = (types.float64, types.float32, types.int32,
```

Comment:
Correct.  Each of these atomic functions have to be supported in hardware (the speculation is that support even needs to be present in the on-chip memory controller), so adding support for types happens very slowly.  Full support for 64-bit only happened in the last few architecture generations.